src/agentscope/model/_zhipu_model.py

```diff
@@ -0,0 +1,394 @@
+        **kwargs: Any,
+    ) -> ChatResponse | AsyncGenerator[ChatResponse, None]:
+        """Get the response from Zhipu AI chat completions API by the given
+        arguments.
+
+        Args:
+            messages (`list[dict]`):
+                A list of dictionaries, where `role` and `content` fields are
+                required, and `name` field is optional.
+            tools (`list[dict]`, default `None`):
+                The tools JSON schemas that the model can use.
+            tool_choice (`Literal["auto", "none", "any", "required"] | str \
+                | None`, default `None`):
+                Controls which (if any) tool is called by the model.
+                Can be "auto", "none", or specific tool name.
+            structured_model (`Type[BaseModel] | None`, default `None`):
+                A Pydantic BaseModel class that defines the expected structure
+                for the model's output.
+            **kwargs (`Any`):
+                The keyword arguments for Zhipu AI chat completions API,
+                e.g. `temperature`, `max_tokens`, `top_p`, etc. Please refer to
+                the Zhipu AI API documentation for more details.
+
+        Returns:
+            `ChatResponse | AsyncGenerator[ChatResponse, None]`:
+                The response from the Zhipu AI chat completions API.
+        """
+
+        # Validate messages
+        if not isinstance(messages, list):
+            raise ValueError(
+                "Zhipu AI `messages` field expected type `list`, "
+                f"got `{type(messages)}` instead.",
+            )
+        if not all("role" in msg and "content" in msg for msg in messages):
+            raise ValueError(
+                "Each message in the 'messages' list must contain a 'role' "
+                "and 'content' key for Zhipu AI API.",
+            )
+
+        kwargs = {
+            "model": self.model_name,
+            "messages": messages,
+            "stream": self.stream,
+            **self.generate_kwargs,
+            **kwargs,
+        }
+
+        if tools:
+            kwargs["tools"] = self._format_tools_json_schemas(tools)
+
+        if tool_choice:
+            self._validate_tool_choice(tool_choice, tools)
+            kwargs["tool_choice"] = self._format_tool_choice(tool_choice)
+
+        if structured_model:
+            if tools or tool_choice:
+                logger.warning(
+                    "structured_model is provided. Both 'tools' and "
+                    "'tool_choice' parameters will be overridden and "
+                    "ignored. The model will only perform structured output "
+                    "generation without calling any other tools.",
+                )
+            # Convert BaseModel to JSON schema for response format
+            kwargs.pop("stream", None)
+            kwargs.pop("tools", None)
+            kwargs.pop("tool_choice", None)
+            kwargs["response_format"] = structured_model
```

Comment:
Since other model APIs either natively support executing specific tools or can guarantee structured output, the `structured_model` parameter always works successfully. However, this API cannot guarantee structured output, so perhaps we should consider raising an error directly in the `__call__` function when structured_model is passed in? cc @DavdGao